susemanager/src/mgr-create-bootstrap-repo

```diff
@@ -301,104 +433,158 @@ def create_repo(label, flush, additional=[]):
             codename="bootstrap"
             reprepro_conf = reprepro_conf_tmpl.substitute(origin="mgr",
                     label="mgr", codename=codename, arches="amd64 i386", comps="main", desc="Bootstrap repo")
-            reprepro_conf_dir = os.path.join(destdir, "conf")
+            reprepro_conf_dir = os.path.join(destdirtmp, "conf")
             if not os.path.exists(reprepro_conf_dir):
                 os.makedirs(reprepro_conf_dir)
             with open(os.path.join(reprepro_conf_dir, "distributions"), "w") as conf_file:
                 conf_file.write(reprepro_conf)
-            logging.debug("Created reprepro config file in {0}".format(os.path.join(destdir, "distributions")))
+            log("Created reprepro config file in {0}".format(os.path.join(destdirtmp, "distributions")), 2)
             debs = [os.path.join(debs_dir, f) for f in os.listdir(debs_dir)]
             try:
-                subprocess.run(["/usr/bin/reprepro", "-b", destdir, "includedeb", codename] + debs, check=True)
-                logging.debug("Removing directory {0}".format(debs_dir))
+                subprocess.run(["/usr/bin/reprepro", "-b", destdirtmp, "includedeb", codename] + debs, check=True)
+                log("Removing directory {0}".format(debs_dir), 2)
                 shutil.rmtree(debs_dir, ignore_errors=True)
             except subprocess.CalledProcessError as err:
-                print("Error creating bootstrap repo.")
-                logging.debug(err)
-                sys.exit(1)
-        else:    
-            os.system("createrepo -s sha %s" % destdir)
+                log_error("Error creating bootstrap repo.")
+                log(err, 2)
+                return 1
+        else:
+            os.system("createrepo -s sha %s" % destdirtmp)
+        # move tmp dir to final location
```

Comment:
In the end this is to generate the repo first, replace the old then. Or am I missing anything?

Reply:
Right and do this as quick as possible.
So we rename the existing to ".old" first. Then the newly created ".tmp" to the really used name. At this point it is "life". Now we have time to remove the ".old" repo as it is not needed anymore.